onnxruntime/core/optimizer/layer_norm_fusion.cc

```diff
@@ -194,24 +201,25 @@ Status LayerNormFusion::ApplyImpl(Graph& graph, bool& modified, int graph_level,
     }
     nodes_to_remove.push_back(pow_node);
 
-    // check if Cast node exists between sub and pow
+    // check if Cast node exists: either between sub and pow, or as second input to pow
     const Node* p_cast_node = graph_utils::FirstParentByType(pow_node, "Cast");
     if (p_cast_node != nullptr) {
       Node& cast_node = *graph.GetNode(pow_node.InputNodesBegin()->Index());
       if (!graph_utils::IsSupportedOptypeVersionAndDomain(cast_node, "Cast", {9, 13}) ||
           cast_node.GetExecutionProviderType() != cast_node.GetExecutionProviderType() ||
```

Comment:
>cast_node [](start = 50, length = 9)

should be reduce_mean_node - happen to notice this when working on the other fusion

Reply:
I am not sure if that's right. The Casts here can either be between sub and pow or as second input to Pow. Will sync offline.